### PR TITLE
spin off Weaken, Reindex files. Implement a usable version of reindex

### PR DESCRIPTION
**Pull request by maxsnew**
Attempting to get actually useful versions of `reindex`, `∃F`, etc. Not ready yet, but take a look at `reindex'` in `Cubical/Categories/Displayed/Constructions/Reindex.agda`. If someone wants to hack a macro for me as I describe there I'd appreciate it.

### CODE REVIEW

File: Cubical/Categories/Displayed/Constructions/Slice.agda [hunk 1]
```diff
@@ -25,29 +37,50 @@ open Category
...
+    open Categoryᴰ
+    test : ∀ {c} → _/C_ .ob[_] c ≡ (Σ[ c' ∈ C .ob ] C [ c , c' ] × Cᴰ .ob[_] c')
+    test = refl
+
+  Δ/C : Functorᴰ Id Cᴰ _/C_
+  Δ/C = mk∫ᴰFunctorᴰ _ Id (Wk.intro Id Id)
+    (BPᴰ.intro _
+      {!!} 
+      -- Fᴰ 
+      -- (reindF (Functor≡ (λ _ → refl) (λ _ → refl))
```
Comment (stschaef) [inline Cubical/Categories/Displayed/Constructions/Slice.agda:56]: When defining it, don't you say not to use `reindF`

File: Cubical/Categories/Displayed/Magmoid.agda [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  infixr 9 _⋆ᴰ_
+  -- infixr 9 _∘ᴰ_
+module _ {C : Category ℓC ℓC'}
+         (Cᴰ : Categoryᴰ C ℓCᴰ ℓCᴰ')
+         where
+  private
+    module C = Category C
+    module Cᴰ = Categoryᴰ Cᴰ
+
+  module _ (idᴰ' : singl {A = ∀ {x} {p : Cᴰ.ob[ x ]} → Cᴰ.Hom[ C.id ][ p , p ]} Cᴰ.idᴰ)
```
Comment (stschaef) [inline Cubical/Categories/Displayed/Magmoid.agda:37]: I don't think I really see what this contractibility is doing here

File: Cubical/Categories/Displayed/Constructions/Reindex.agda [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                → isProp (f Equality.≡ g)
+    isPropEqHom {f = f}{g} =
+      subst isProp (Equality.PathPathEq {x = f}{y = g}) (C.isSetHom f g)
+
+  open Categoryᴰ Cᴰ
+
+  reind' : {a b : C.ob} {f g : C [ a , b ]} (p : f Equality.≡ g)
+      {aᴰ : ob[ a ]} {bᴰ : ob[ b ]}
+    → Hom[ f ][ aᴰ , bᴰ ] → Hom[ g ][ aᴰ , bᴰ ]
+  reind' p = Equality.transport Hom[_][ _ , _ ] p
```
Comment (stschaef) [inline Cubical/Categories/Displayed/Constructions/Reindex.agda:50]: I have some vague vibes to why using the identity type might be better than the path type, but what does this actually gain over the existing `reind`

File: Cubical/Categories/Displayed/Magmoid.agda (hunk 2 above)
Comment (maxsnew) [reply Cubical/Categories/Displayed/Magmoid.agda:37]: I don't understand the question

File: Cubical/Categories/Displayed/Constructions/Reindex.agda (hunk 3 above)
Comment (maxsnew) [reply Cubical/Categories/Displayed/Constructions/Reindex.agda:50]: see Quotient.More. I was able to replace an ad hoc construction with this new `reind'` without transport hell

**Pull request merged**